Add support for JS configs

claytercek opened the pull request:
fixes #70

- adjusted ConfigManager to allow loading of js configs
- made the CLI `--config` flag compatible with js configs
- slightly adjusted the config loading procedure:
   - only load first found config if no config is passed via CLI args
   - fatally errors our if it is unable to load first config
- added `launchpad.config.js` and `launchpad.config.mjs` as default config files and prioritized them when searching
- added deprecation warning when using JSON configs

Tested on a couple projects, and this seems to work! Importing from other files in the js config works too. I'm breaking config intellisense out into another issue.

I also adjusted the config loading procedure, as I think it makes more sense to fail rather than keep searching for valid configs when a broken config is found.

ref #65

claytercek commented:
Oh also for some type-safety updates coming down the line, I made it so that ConfigManager is no longer a singleton, and is instantiated and passed down when launchpad commands are run from the CLI. This way ConfigManager can work with a generic config shape, that can be defined when the manager is instantiated. This'll help with compartmentalizing the config shapes and colocating them with their respective sub-packages (ie launchpad content will own the content config shape, and launchpad core will incorporate it into the larger config).

benjaminbojko commented on packages/utils/lib/config-manager.js:
```diff
@@ -133,58 +114,74 @@ export class ConfigManager {
+	 * @returns {string | null} The absolute path to the file or null if it doesn't exist.
+	 * @private
+	 */
+	static _fileExistsRecursive(filePath) {
```
Minor comment: Given this name I would expect this to return a `boolean`. How about something like `_findFirstFileRecursive()`?

claytercek replied:
Good call! Updated

benjaminbojko reviewed (commented):
Overall this looks great! Thanks for digging so deep and cleaning all those messy typedefs (sheesh, I thought the existing lint and jsdoc config would catch that, but so many small mistakes 😨). 

I couldn't find an example config in this PR. Could you point me to one or post one in a README somewhere? Primarily curious about how you'd export/define the object to import in the js file.

Maybe we should hold pending #125 ?

claytercek commented:
@benjaminbojko I just updated the README to reflect support for js configs, and modified the example config. Right now it just looks for the default export from the config file:

https://github.com/bluecadet/launchpad/blob/3dfa8c76087362bcd2a957eb5fa5f8f9d72f7626/packages/launchpad/README.md?plain=1#L49-L72

Yeah I think we can merge to develop but hold on releasing until we have #125 and maybe #116 too?

benjaminbojko commented:
Great, thank you! I think the default export makes the most sense. I'll review the other PRs asap.

@claytercek can you merge this one in?
